internal/services/azapi_resource_data_source.go

```diff
@@ -26,15 +26,28 @@ func ResourceAzApiDataSource() *schema.Resource {
 
 		Schema: map[string]*schema.Schema{
 			"name": {
-				Type:         schema.TypeString,
-				Required:     true,
-				ValidateFunc: validation.StringIsNotEmpty,
+				Type:          schema.TypeString,
+				Optional:      true,
+				ValidateFunc:  validation.StringIsNotEmpty,
+				RequiredWith:  []string{"parent_id"},
+				ConflictsWith: []string{"resource_id"},
+				AtLeastOneOf:  []string{"name", "resource_id"},
```

Comment:
This can be merged into: `ExactlyOneOf`?

Reply:
Good catch, I'll update it.

Reply:
The ConfclitsWith can also be removed than.

Reply:
Yes, and it has been removed.